### PR TITLE
[RSDK-6871] add edited map to slam client

### PR DESCRIPTION
**Pull request by JohnN193**
ticket: https://viam.atlassian.net/browse/RSDK-6871

this pr adds the ReturnEditedMap bool to the slam typescript client. Will not merge until after the API PR is merged and this PR is updated accordingly
api change PR: https://github.com/viamrobotics/api/pull/456

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -55,7 +55,7 @@ update-buf: $(node_modules)
 .PHONY: build-buf
 build-buf: $(node_modules) clean-buf
 	$(buf) generate $$(./scripts/get-buf-lock-version.js buf.build/googleapis/googleapis)
-	$(buf) generate $$(./scripts/get-buf-lock-version.js buf.build/viamrobotics/api) --path common,component,robot,service,app
```
Comment (JohnN193) [inline Makefile:58]: will revert before merging

**Comment by purplenicole730** [comment]
If this isn't ready to be merged in and will be updated in the future, please set this as a draft PR (bottom of the reviewers section on the right)

**Comment by JohnN193** [comment]
> Is this ready to be reviewed now? If so, I will happily review! If this isn't ready and will be updated in the future, please set this as a draft PR (bottom of the reviewers section on the right)

Yep! I wanted to get this PR reviewed before merging in my [api change](https://github.com/viamrobotics/api/pull/456)

**Comment by njooma** [comment]
> Also just to confirm, once I merge my api pr do I need to do anything to get the `Automated Protos Update` to run?

Nope! That PR will happen automatically, you don't need to do anything

**Comment by JohnN193** [comment]
Added a test for the parameter!

**Review by purplenicole730 — APPROVED** [review]
Looks good! Thanks for adding the test to mock the request

**Pull request merged**